tests/integration/conftest.py

```diff
@@ -57,36 +57,37 @@ def get_report_files():
 def pytest_collection_modifyitems(session, config, items):
     selected_tests = []
     deselected_tests = []
+    if not global_parameters.avoid_platform_based_deselection:
+        for item in items:
+            supported_platforms = PLATFORMS.intersection(mark.name for mark in item.iter_markers())
+            plat = sys.platform
 
-    for item in items:
-        supported_platforms = PLATFORMS.intersection(mark.name for mark in item.iter_markers())
-        plat = sys.platform
-
-        selected = True
-        if supported_platforms and plat not in supported_platforms:
-            selected = False
-
-        host_type = 'agent' if 'agent' in get_service() else 'server'
-        supported_types = HOST_TYPES.intersection(mark.name for mark in item.iter_markers())
-        if supported_types and host_type not in supported_types:
-            selected = False
-        # Consider only first mark
-        levels = [mark.kwargs['level'] for mark in item.iter_markers(name="tier")]
-        if levels and len(levels) > 0:
-            tiers = item.config.getoption("--tier")
-            if tiers is not None and levels[0] not in tiers:
-                selected = False
-            elif item.config.getoption("--tier-minimum") > levels[0]:
```

Comment:
Great! I'll add a comment explaining a little bit to add more clarity.

Reply:
Done bb079ab